### PR TITLE
Implement purchase availability

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds information if a purchase is not available or when it will be available on product details page.

### Screenshots

Zrzut ekranu 2020-09-1 o 11 27 35
Zrzut ekranu 2020-09-1 o 11 16 11

### API branch to test with
add-searchable-and-available-for-purchase-flags-to-product

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://add-searchable-and-available-for-purchase-flags-to-product.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by orzechdev** [comment]
Locally e2e tests pass
Zrzut ekranu 2020-08-27 o 15 17 51

**Comment by patrys** [comment]
Task linked: [CU-7enmfm Storefront changes for purchase availability date.](https://app.clickup.com/t/7enmfm)

**Review by AlicjaSzu — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "isomorphic-fetch": "^2.2.1",
     "js-base64": "^2.4.8",
     "lodash": "^4.17.19",
+    "moment": "^2.27.0",
```
Comment (dominik-zeglen) [inline package.json:43]: No need for `moment`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Comment by orzechdev** [comment]
Do not merge if backend is not merged https://github.com/mirumee/saleor/pull/6060

**Review by dominik-zeglen — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**